### PR TITLE
Switch next

### PR DESCRIPTION
**Pull request by AJenbo**
This will enable custom dialogs and use the now standardized NONET option meaning almost all Switch specific exceptions are gone.

### CODE REVIEW

File: SourceX/DiabloUI/diabloui.cpp [hunk 1]
```diff
@@ -78,6 +83,9 @@ void UiInitList(int min, int max, void (*fnFocus)(int value), void (*fnSelect)(i
 	SDL_StopTextInput(); // input is enabled by default
 	for (int i = 0; i < itemCnt; i++) {
 		if (items[i].type == UI_EDIT) {
+#ifdef SWITCH
+			switch_start_text_input(items[i-1].caption, items[i].caption, 0);
```
Comment (glebm) [inline SourceX/DiabloUI/diabloui.cpp]:
```c++
switch_start_text_input(
  items[i - 1].art_text.text, items[i].edit.value, /*multiline=*/0);
```

File: SourceX/main.cpp [hunk 2]
```diff
@@ -25,5 +29,20 @@ static std::string build_cmdline(int argc, char **argv)
...
+#ifdef SWITCH
+	// enable network and stdio on Switch
+	socketInitializeDefault();
+	// enable error messages via nxlink on Switch
+	nxlink_sock = nxlinkStdio();
+#endif
+
 	return dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);
+
+#ifdef SWITCH
```
Comment (mmaret) [inline SourceX/main.cpp]: code unreachable

Comment (rsn8887) [reply SourceX/main.cpp]: One could simply do ```
int return_value = dvl::WinMain(NULL, NULL, (char *)cmdline.c_str(), 0);
#ifdef SWITCH
// disable network and stdio on Switch
 	if (nxlink_sock != -1)
 		close(nxlink_sock);
 	socketExit();
 #endif

return return_value;
...

Comment (AJenbo) [reply SourceX/main.cpp]: yep

**Pull request closed without merging**